### PR TITLE
Feature/add default iamidentities support

### PR DESCRIPTION
**Pull request by levtomer66**
* Organized IAM identity mappings into separate files for default and additional identities.
* default-iamidentity-mapping.yaml for the default IAM identity.
* additional-iamidentity-mapping.yaml for additional IAM identities.

Adjusted values.yaml for default configuration.

### CODE REVIEW

File: charts/rustrial-aws-eks-iam-auth-controller/values.yaml [hunk 1]
```diff
@@ -67,3 +67,17 @@ tolerations: []
 affinity: {}
 
 topologySpreadConstraints: {}
+
+defaultIdentity:
+  enabled: false
+  roleArn: "arn:aws:iam::123456789012:role/example-role"
+
+additionalIdentities: {}
```
Comment (alex-berger) [inline charts/rustrial-aws-eks-iam-auth-controller/values.yaml]: This is an array not an object and thus its default value must be `[]` and not `{}`.

File: charts/rustrial-aws-eks-iam-auth-controller/values.yaml [hunk 2]
```diff
@@ -67,3 +67,17 @@ tolerations: []
 affinity: {}
 
 topologySpreadConstraints: {}
+
+defaultIdentity:
```
Comment (alex-berger) [inline charts/rustrial-aws-eks-iam-auth-controller/values.yaml]: @levtomer66 IMHO, we should not add the `defaultIdentity` part and only focus on the `additionalIdentities` part. The `defaultIdentity` is too specific and the same functionality can be achieved using `additionalIdentities`, for example like this: 

```yaml
additionalIdentities:
  - name: "aws-ec2-nodes"
    arn: "arn:aws:iam::{{AccountID}}:role/my-eks-node-instance-profile"
    username: "system:node:{{EC2PrivateDNSName}}"
    groups:
    - system:bootstrappers
    - system:nodes
```

If we want to make this more obvious to end-users we can kind of document it like this in `values.yaml`:

```yaml
additionalIdentities: []
#  - name: "aws-ec2-nodes"
#    arn: "arn:aws:iam::{{AccountID}}:role/my-eks-node-instance-profile"
#    username: "system:node:{{EC2PrivateDNSName}}"
#    groups:
#    - system:bootstrappers
#    - system:nodes
```

File: charts/rustrial-aws-eks-iam-auth-controller/templates/additional-iamidentity-mapping.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{{- range .Values.additionalIdentities }}
+apiVersion: iamauthenticator.k8s.aws/v1alpha1
+kind: IAMIdentityMapping
+metadata:
+  name: {{ .name }}
+  annotations:
+    "helm.sh/hook": post-install
```
Comment (alex-berger) [inline charts/rustrial-aws-eks-iam-auth-controller/templates/additional-iamidentity-mapping.yaml]: Can you please elaborate why it is necessary to create those resources in a "post-install" hook?

File: charts/rustrial-aws-eks-iam-auth-controller/templates/default-iamidentity-mapping.yaml
Comment (alex-berger) [inline charts/rustrial-aws-eks-iam-auth-controller/templates/default-iamidentity-mapping.yaml:1]: Let's remove that file in favor of `additionalIdentities` (see https://github.com/rustrial/aws-eks-iam-auth-controller/pull/46#discussion_r1429132953).

**Review by alex-berger — CHANGES_REQUESTED** [review]

File: charts/rustrial-aws-eks-iam-auth-controller/values.yaml (hunk 2 above)
Comment (levtomer66) [reply charts/rustrial-aws-eks-iam-auth-controller/values.yaml]: Sure, fixed 👍🏼

File: charts/rustrial-aws-eks-iam-auth-controller/templates/additional-iamidentity-mapping.yaml (hunk 3 above)
Comment (levtomer66) [reply charts/rustrial-aws-eks-iam-auth-controller/templates/additional-iamidentity-mapping.yaml]: We want the CRD to present first to prevent failures

Comment (alex-berger) [reply charts/rustrial-aws-eks-iam-auth-controller/templates/additional-iamidentity-mapping.yaml]: AFAIK, CRDs (in `/crds/`) are always first installed by Helm before applying any other resources, no need to use hooks for that purpose. See also https://helm.sh/docs/topics/charts_hooks/.

Comment (levtomer66) [reply charts/rustrial-aws-eks-iam-auth-controller/templates/additional-iamidentity-mapping.yaml]: OK great I'm removing this

**Pull request merged**